### PR TITLE
#3044337: Configurable landing point on profile 

### PR DESCRIPTION
**Pull request by robertragas**
## Problem
Currently when going to a user you always end up at the stream page. Some customers would like them to end up at the user profile.

## Solution
Added the possibility to set a configuration item to land on either the information page or stream.

Right now it has the two options hardcoded but can be extended in the future by getting the user tabs dynamic based on permissions set for a client (since it might be that users are not allowed to view a page).

## Issue tracker
https://www.drupal.org/project/social/issues/3044337

## How to test
- [ ] drush updb or install
- [ ] Go to configuration -> open social -> user settings
- [ ] By default it lands on stream, but try to alter it to Information
- [ ] Click on a user from a stream link like mentions or username, you will end up at the profile information.
- [ ] Try to go to your own profile through the account header item, this still goes to your stream page.

## Release notes
We have added a new setting where you can configure the landing page when visiting a user's profile. This can be set under configuration -> open social -> user settings.

## Change record
https://www.drupal.org/node/3044338

### CODE REVIEW

**Comment by robertragas** [comment]
Made a fix so tests in travis should also go green now.

**Comment by jochemvn** [comment]
@robertragas 
The permission to change the entry point on user profile is not set for any user... I assume a SM needs to be able to do this?

**Comment by robertragas** [comment]
@jochemvn merged in the base branch and updated the install script with the permission so site managers can set the configuration.

File: modules/social_features/social_user/social_user.install [hunk 1]
```diff
@@ -126,6 +126,15 @@ function social_user_update_8008() {
   // Create the default configuration for the social_user default landing page.
   $config = \Drupal::getContainer()->get('config.factory')->getEditable('social_user.settings');
   $config->set('social_user_profile_landingpage', 'entity.user.canonical')->save();
+
+  // Add permissions for the role.
+  $rid = 'sitemanager';
```
[the remark this answers is not in the document]
Comment (robertragas) [reply modules/social_features/social_user/social_user.install]: done

File: modules/social_features/social_user/social_user.install [hunk 2]
```diff
@@ -126,6 +126,15 @@ function social_user_update_8008() {
...
+
+  $role = Role::load($rid);
+  if ($role) {
```
[the remark this answers is not in the document]
Comment (robertragas) [reply modules/social_features/social_user/social_user.install]: done

**Pull request merged**